src/connectors/search-box/__tests__/connectSearchBox-test.js

```diff
@@ -377,105 +399,89 @@ See documentation: https://www.algolia.com/doc/api-reference/widgets/search-box/
     });
   });
 
-  describe('routing', () => {
-    const getInitializedWidget = (config = {}) => {
-      const renderFn = jest.fn();
-      const makeWidget = connectSearchBox(renderFn);
-      const widget = makeWidget({
-        ...config,
+  describe('getWidgetState', () => {
+    test('should give back the object unmodified if the default value is selected', () => {
+      const [widget, helper] = getInitializedWidget();
+      const uiStateBefore = {};
+      const uiStateAfter = widget.getWidgetState(uiStateBefore, {
+        searchParameters: helper.state,
+        helper,
       });
+      expect(uiStateAfter).toBe(uiStateBefore);
+    });
 
-      const initialConfig = {};
-      const helper = algoliasearchHelper({}, '', initialConfig);
-      helper.search = jest.fn();
-
-      widget.init({
+    test('should add an entry equal to the refinement', () => {
+      const [widget, helper, refine] = getInitializedWidget();
+      refine('some query');
+      const uiStateBefore = {};
+      const uiStateAfter = widget.getWidgetState(uiStateBefore, {
+        searchParameters: helper.state,
         helper,
-        state: helper.state,
-        createURL: () => '#',
       });
+      expect(uiStateAfter).toMatchSnapshot();
+    });
 
-      const { refine } = renderFn.mock.calls[0][0];
-
-      return [widget, helper, refine];
-    };
-
-    describe('getWidgetState', () => {
-      test('should give back the object unmodified if the default value is selected', () => {
-        const [widget, helper] = getInitializedWidget();
-        const uiStateBefore = {};
-        const uiStateAfter = widget.getWidgetState(uiStateBefore, {
+    test('should give back the same instance if the value is alreay in the uiState', () => {
+      const [widget, helper, refine] = getInitializedWidget();
+      refine('query');
+      const uiStateBefore = widget.getWidgetState(
+        {},
+        {
           searchParameters: helper.state,
           helper,
-        });
-        expect(uiStateAfter).toBe(uiStateBefore);
+        }
+      );
+      const uiStateAfter = widget.getWidgetState(uiStateBefore, {
+        searchParameters: helper.state,
+        helper,
       });
+      expect(uiStateAfter).toBe(uiStateBefore);
```

Comment:
Do we care about getting the same instance? We just want to have the same content, right?

Reply:
if it's ever updated we can change this assertion I guess?